compose/service.py

```diff
@@ -332,6 +332,18 @@ def _next_container_number(self, all_containers):
         numbers = [parse_name(c.name).number for c in all_containers]
         return 1 if not numbers else max(numbers) + 1
 
+    def _get_dns_resolved(self, dns):
+        if type(dns) != list:
+            dns = [dns]
+        containers = dict([(link_name, service.containers()) for service, link_name in self.links])
+        dns_resolved = []
+        for d in dns:
+            if d in containers.keys():
+                dns_resolved.extend([c.get('NetworkSettings.IPAddress') for c in containers[d]])
+            else:
+                dns_resolved.append(d)
+        return dns_resolved
```

Comment:
It seems like this doesn't need to be restricted to just linked containers. It should be able to use *any* container with the appropriate name (maybe a container that isn't even part of the composition).

If this were to use `container:<name>` syntax as I've suggested, you should be able to extract a lot of shared code from [project.get_net()](https://github.com/docker/compose/blob/9931cd2c4cd6683cef5ded0d8f15c71c7a9f7741/compose/project.py#L161).

The "lookup container first by service name, then by container name" would be identical (which is great for consistency).